src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  margin: 0 auto;
+  gap: $gap;
+  padding: $padding;
+  width: calendarWidth(
+    $day-size,
+    $colums-count,
+    $gap
+  );
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    line-height: 100px;
+    text-align: center;
+    font-size: 30px;
+    border: 1px solid black;
+    background-color: #eee;
+    transition: 0.5s;
+
+    @for $index from 1 through 31 {
+      &:nth-child(n + #{$index})::before {
+        content: "#{$index}";
+      }
+
+      &:hover {
+        cursor: pointer;
+        background-color: pink;
+        transform: translateY(-20px);
+      }
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length-#{$index}
+    :nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  @each $day, $num in $day-value {
+    &--start-day-#{$day} :first-child {
+      margin-left: $num * $day-size + $num * $gap;
```

Comment:
you can create a function for this)

Reply:
Yes, but I wanted to try 'each' =))
Or am I not?